src/main/java/io/appform/dropwizard/sharding/utils/TransactionHandler.java

```diff
@@ -25,13 +25,15 @@
 import org.hibernate.Transaction;
 import org.hibernate.context.internal.ManagedSessionContext;
 import org.hibernate.resource.transaction.spi.TransactionStatus;
+import org.jboss.logging.MDC;
```

Comment:
jboss MDC is available in core hibernate dependecy tree and did not want to introduce anything new. SInce the functionally there is no difference as such for this particular usecase; I used the one that was readily available. Let me know if adding another dependency is ok. Will just replace the dependency. This was done to enable multitenancy support w.t.t encryption. Hibernate type registry is singleton and is not reentrant. I had to workaround by setting the context without breaking the functionality when used in non-multi tenancy configuration.

Reply:
Ack. Makes sense. No need for additional lib given that usage is constraint within the lib